GildedRoseKata/ItemAlter.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+
+namespace GildedRoseKata
+{
+    class ItemAlter
+    {
+        public virtual void UpdateItemQuality(Item item)
+        {
+            Console.WriteLine("-------- day " + item + " --------");
+            Console.WriteLine("name, sellIn, quality");
+            throw new Exception("Unknown Item Type");
+        }
+
+        public static Category GetCategory(Item item)
+        {
+            if (item.Name.Equals("Aged Brie"))
+                return Category.AgedBrie;
+            else if (item.Name.Equals("Backstage passes"))
+                return Category.BackstagePasses;
+            else if (item.Name.Equals("Sulfuras"))
+                return Category.LegendaryItem;
+            else if (item.Name.Equals("Conjured"))
+                return Category.ConjuredItem;
+            else
+                return Category.NormalItem;
+        }
+
+        public static ItemAlter GetInstanceFor(Item item)
+        {
+            if (GetCategory(item) == Category.AgedBrie)
+                return new AgedBrieAlter();
+            else if (GetCategory(item) == Category.BackstagePasses)
+                return new BackstagePassAlter();
+            else if (GetCategory(item) == Category.NormalItem)
+                return new NormalItemAlter();
+            else if (GetCategory(item) == Category.LegendaryItem)
+                return new LegendaryItemAlter();
+            else if (GetCategory(item) == Category.ConjuredItem)
+                return new ConjuredItemAlter();
+            else
+                return new ItemAlter();
+        }
```

Comment:
Pattern matching

GetCategory is only used here, the ItemAlter to use should be computed directly

Last else will never be executed according to the GetCategory function implementation